src/IO/S3/getObjectInfo.cpp

```diff
@@ -91,11 +91,27 @@ ObjectInfo getObjectInfo(
     }
     if (throw_on_error)
     {
-        throw S3Exception(
-            error.GetErrorType(),
-            "Failed to get object info: {}. HTTP response code: {}",
-            error.GetMessage(),
-            static_cast<size_t>(error.GetResponseCode()));
+        auto response_code = static_cast<size_t>(error.GetResponseCode());
+        auto error_type = error.GetErrorType();
+        auto error_message = error.GetMessage();
+
+        String dynamic_error_message;
+        if (error_type == Aws::S3::S3Errors::ACCESS_DENIED)
+        {
+            dynamic_error_message = fmt::format(
+                "Access denied to object info: {}. HTTP response code: {}. Error type: {}. "
+                "Please make sure that the provided AWS credentials are correct and have sufficient permissions.",
```

Comment:
Can we mention here from which source used credentials actually came from?
It may not be obvious if there is a `.aws/credentials` present, but IMDS keys were used.